### PR TITLE
Requirements with tools and components

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
# What was changed:

## C++ part:

- allowed requirements with both components and tools/qualities.

## JSON part:

- added `solder` to components of `soldering_standard` requirement;
- added `filament` to components of `sewing_standard` requirement;
- simplified multiple recipes using updated requirements (removed duplicate qualities and such).

# Notes:

- will probably split such big changes to multiple PRs;
- this change requires heavy in-game testing.

### CODE REVIEW

File: data/json/recipes/armor/legs.json [hunk 1]
```diff
@@ -449,7 +425,7 @@
     "skills_required": [ "fabrication", 5 ],
     "time": 70000,
     "autolearn": true,
-    "using": [ [ "filament", 100 ] ],
+    "using": [ [ "sewing_standard", 100 ] ],
     "qualities": [
       { "id": "CUT", "level": 1 },
       { "id": "SEW", "level": 1 }
```
Comment (nexusmrsep) [inline data/json/recipes/armor/legs.json]: ommit here?

File: data/json/recipes/armor/legs.json [hunk 2]
```diff
@@ -498,7 +472,7 @@
     "skill_used": "tailor",
     "time": 30000,
     "autolearn": true,
-    "using": [ [ "filament", 12 ] ],
+    "using": [ [ "sewing_standard", 12 ] ],
     "qualities": [
       { "id": "CUT", "level": 1 },
       { "id": "SEW", "level": 1 }
```
Comment (nexusmrsep) [inline data/json/recipes/armor/legs.json]: ommit here?

File: data/json/recipes/armor/legs.json [hunk 3]
```diff
@@ -527,7 +500,7 @@
     "skill_used": "tailor",
     "time": 20000,
     "autolearn": true,
-    "using": [ [ "filament", 10 ] ],
+    "using": [ [ "sewing_standard", 10 ] ],
     "qualities": [
       { "id": "CUT", "level": 1 },
       { "id": "SEW", "level": 1 }
```
Comment (nexusmrsep) [inline data/json/recipes/armor/legs.json]: same here i guess

**Comment by nexusmrsep** [comment]
Actualy makes me wondering about why there was only one level of sewing tools (`"SEW", "level": 1` everywhere)

**Comment by ZhilkinSerg** [comment]
> Actualy makes me wondering about why there was only one level of sewing tools ("SEW", "level": 1 everywhere)

There are no items with `SEW` quality of another level.

**Comment by ZhilkinSerg** [comment]
> Isn.t a wooden or bone needle worse then a steel one?

There is really a little difference. Yes, steel needle would be more sturdy, but you can sew with bone needle too.

**Comment by nexusmrsep** [comment]
Maybe, bur consider sewing leather, boots for example. Or hard leather armor. Finger tab would break the wooden or bone needle sooner then later. You would even use metal awl to get throu if possible. Yes this can be done with wooden needle or bone needle too, but steel needle might be faster. But thats a material for a different discussion elsewhere i guess.

**Comment by ZhilkinSerg** [comment]
> Maybe, but consider sewing leather, boots for example. Or hard leather armor. Finger tab would break the wooden or bone needle sooner then later. You would even use metal awl to get throu if possible. Yes this can be done with wooden needle or bone needle too, but steel needle might be faster. But thats a material for a different discussion elsewhere i guess.

Makes sense. Expanding requirements mechanics is a good idea for future PRs.

**Pull request merged**